netbox/extras/models/scripts.py

```diff
@@ -78,6 +121,35 @@ def _get_name(cls):
 
         return scripts
 
+    def sync_classes(self):
+        db_classes = {}
+        for obj in self.scripts.filter(module=self):
+            db_classes[obj.name] = obj
+
+        db_classes_set = {k for k in db_classes.keys()}
+
+        module_scripts = self.get_module_scripts
+
+        module_classes_set = {k for k in module_scripts.keys()}
+
+        # remove any existing db classes if they are no longer in the file
+        removed = db_classes_set - module_classes_set
+        for name in removed:
+            db_classes[name].delete_if_no_jobs()
+
+        added = module_classes_set - db_classes_set
+        for name in added:
+            Script.objects.create(
+                module=self,
+                name=name,
+                is_executable=True,
+            )
+
+    def sync_data(self):
+        super().sync_data()
+        self.sync_classes()
+
     def save(self, *args, **kwargs):
         self.file_root = ManagedFileRootPathChoices.SCRIPTS
-        return super().save(*args, **kwargs)
+        super().save(*args, **kwargs)
```

Comment:
Be careful to still return the result of calling `save()`. We might need to invoke `sync_classes()` via the `post_save` signal.